### ISSUE TITLE
google_compute_router_peer peer_asn and advertise_mode forcing replacement

### ISSUE THREAD

**Issue by tnguyen14** [hashicorp/terraform-provider-google#7977]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

0.12.29

### Affected Resource(s)

* google_compute_router_peer

### Terraform Configuration Files

```tf
# Copy-paste your Terraform configurations here.
#
# For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file.
# For security, you can also encrypt the files using our GPG public key:
#    https://www.hashicorp.com/security
#
# If reproducing the bug involves modifying the config file (e.g., apply a config,
# change a value, apply the config again, see the bug), then please include both:
# * the version of the config before the change, and
# * the version of the config after the change.
```

### Expected Behavior

We have BGP peers that are of type "MANAGED_BY_ATTACHMENT". We use `terraform import` to get them managed by Terraform.

On the GCP console, the "Advertisement Mode" for the peers are shown as "Custom", however, when imported, the terraform state has the attribute `advertise_mode` as `DEFAULT`.

When we run `terraform plan`, the output shows that the `advertise_mode` needs to be changed from `DEFAULT` to `CUSTOM` (as the terraform configuration file expects), and that the `peer_asn` needs to be updated as well (also as the terraform configuration file dictates).

The changes on these 2 attributes indicate that they would force a replacement of the BGP peer resource.

I don't think that changing these attributes should force a replacement. In the GCP console, I could go in and update these attributes inline without having to delete those resources.

### Steps to Reproduce

1. `terraform apply`

### References

* b/183546073

**Comment by slevenick** [comment]
Hey @tnguyen14 I would be curious why `advertise_mode` is not being read correctly when you import it.

Are you able to include debug logs for your import & plan afterwards?

Otherwise, adding support for updating `peer_asn` and `advertise_mode` would be a feature request and I can add it to the list to be triaged

**Comment by tnguyen14** [comment]
@slevenick thanks for the response. I would say go with the feature request for now. I will open a separate ticket for the import issue with more debug logs.

**Comment by tnguyen14** [comment]
@slevenick I created https://github.com/hashicorp/terraform-provider-google/issues/8009 for the import issue

**Comment by rileykarson** [comment]
`advertisedGroups` mentions:

```
Note that this field can only be populated if advertiseMode is CUSTOM and overrides the list defined for the router (in the "bgp" message). These groups are advertised in addition to any specified prefixes. Leave this field blank to advertise no custom groups.
```

`advertiseMode`'s documentation is a little sparse, it only indicates:

```
User-specified flag to indicate which mode to use for advertisement.
```

What I _suspect_ is that the same is true for this field as well, and that the `bgp` settings override those in the peer. That makes a past assumption of ours- that no setting returned means `"DEFAULT"`- is false. Note that default doesn't mean "default for the router" here, it's the [default advertisement mode](https://cloud.google.com/network-connectivity/docs/router/concepts/overview#route-advertisement-default)

The good news: You can ignore this value. Terraform deciding it has a value of `"DEFAULT"` is entirely clientside, and the field is immutable (eg. changing the value recreates the resource) and as-implemented Terraform is unable to send a bad update. It's completely benign to ignore- I would suggest setting `lifecycle.ignore_changes` on the field.

We'll still want to investigate and fix this behaviour, we'll want to confirm what we do with the service team.

**Comment by ScottSuarez** [comment]
> Note that default doesn't mean "default for the router" here, it's the [default advertisement mode](https://cloud.google.com/network-connectivity/docs/router/concepts/overview#route-advertisement-default)

Default seems to imply inherited [from bpg session.](https://cloud.google.com/network-connectivity/docs/router/concepts/overview#route-advertisement-default) 

```
Each BGP session for a Cloud Router also has a default advertisement. By default, Cloud Router propagates its route advertisements to all of its BGP sessions. If you configure custom route advertisements on a Cloud Router, its BGP sessions inherit those custom advertisements.
```

The current behavior looks okay to me, save the persistent diff. We only need to default if empty and look into seeing if a bpgpeer's AdvertiseMode and AdvertisedGroup is changeable.

**Comment by rileykarson** [comment]
Thanks for looking into this!

**Comment by tnguyen14** [comment]
> The current behavior looks okay to me.

@ScottSuarez are you saying that it's okay to force replacement if `peer_asn` and `advertise_mode` change?

**Comment by ScottSuarez** [comment]
> > The current behavior looks okay to me.
> 
> @ScottSuarez are you saying that it's okay to force replacement if `peer_asn` and `advertise_mode` change?
No I was referring to behavior of `DEFAULT` being set when `bpg.advertisedMode` is `CUSTOM`.

I investigated the updatable fields and I was able to confirm that they are able to be updated. I've put out a pr to enable you do so. Thanks for your help here @tnguyen14 .

**Comment by tnguyen14** [comment]
Does anyone know when the change merged in #8862 will be released?

**Comment by ScottSuarez** [comment]
April 19th it will be released as apart of `3.65.0`
